via/static/scripts/video_player/hooks/use-app-layout.ts

```diff
@@ -0,0 +1,53 @@
+import type { RefObject } from 'preact';
+import { useCallback, useLayoutEffect, useState } from 'preact/hooks';
+
+type AppSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
+
+/**
+ * Watch for resizes on `element` and invoke `onSizeChanged` callback
+ */
+export function observeElementSize(
```

Comment:
If we're relying on `ResizeObserver`'s availability, this helper doesn't do much so it could be inlined into the hook below.